napari/_qt/widgets/qt_extension2reader.py

```diff
@@ -20,94 +30,248 @@ class Extension2ReaderTable(QWidget):
 
     valueChanged = Signal(int)
 
-    def __init__(self, parent=None):
+    def __init__(self, parent=None, npe2_readers=None, npe1_readers=None):
         super().__init__(parent=parent)
 
+        npe2, npe1 = get_all_readers()
+        if npe2_readers is None:
+            npe2_readers = npe2
+        if npe1_readers is None:
+            npe1_readers = npe1
+
+        self._npe2_readers = npe2_readers
+        self._npe1_readers = npe1_readers
+
         self._table = QTableWidget()
         self._table.setShowGrid(False)
+        self._set_up_table()
+        self._edit_row = self._make_new_preference_row()
         self._populate_table()
 
+        instructions = QLabel(
+            trans._(
+                'Enter a filename pattern to associate with a reader e.g. "*.tif" for all TIFF files.'
+                + 'Available readers will be filtered to those compatible with your pattern. Hover over a reader to see what patterns it accepts.'
+                + '\n\nPreference saving for folder readers is not supported, so these readers are not shown.'
+                + '\n\nFor documentation on valid filename patterns, see https://docs.python.org/3/library/fnmatch.html'
+            )
+        )
```

Comment:
I think translations need to be on a single line, so each line needs to be `trans._()`d separately and then concatenated. (Or you could do a big-ass line but I like concatenating the translations better.)